ledger/complete/ledger.go

```diff
@@ -368,3 +375,21 @@ func (l *Ledger) DumpTrieAsJSON(state ledger.State, outputFilePath string) error
 
 	return trie.DumpAsJSON(writer)
 }
+
+// this operation should only be used for exporting
+func (l *Ledger) keepOnlyOneTrie(state ledger.State) error {
+	// don't write things to WALs
+	l.wal.PauseRecord()
+	defer l.wal.UnpauseRecord()
+
+	allTries, err := l.forest.GetTries()
+	if err != nil {
+		return err
+	}
+	for _, trie := range allTries {
+		if !bytes.Equal(trie.RootHash(), ledger.RootHash(state)) {
+			l.forest.RemoveTrie(trie.RootHash())
```

Comment:
calculated RootHash twice. maybe could cache

Reply:
it is actually cached the trie, so this basically reads the same value without any extra overhead. but I'll add extra vars here.